gem/optimise.py

```diff
@@ -413,3 +480,565 @@ def aggressive_unroll(expression):
     expression, = unroll_indexsum((expression,), max_extent=numpy.inf)
     expression, = remove_componenttensors((expression,))
     return expression
+
+
+# ---------------------------------------
+# Count flop of expression, "as it is", no reordering etc
+@singledispatch
+def _count_flop(node, self):
+    raise AssertionError("cannot handle type %s" % type(node))
+
+
+@_count_flop.register(IndexSum)
+def _count_flop_single(node, self):
+    return self(node.children[0])
+
+
+@_count_flop.register(MathFunction)
+@_count_flop.register(LogicalNot)
+@_count_flop.register(LogicalAnd)
+@_count_flop.register(LogicalOr)
+def _count_flop_func(node, self):
+    return self(node.children[0]) + 1
+
+
+@_count_flop.register(Conditional)  # this is not quite right
```

Comment:
Sounds good. Considering code generation, sum is probably closer to reality than max.